bin/pycheck

```diff
@@ -18,7 +18,7 @@ cd "$(dirname "$0")/.."
 . misc/shlib/shlib.bash
 
 pyright_version=$(sh ci/builder/pyright-version.sh)
-typecheck_cmd="npx pyright@$pyright_version"
+typecheck_cmd="npx pyright@$pyright_version --warnings"
```

Comment:
`--warnings` causes the build to fail if there are any warnings. A few warnings are left, [showing that this works.](https://buildkite.com/materialize/tests/builds/62409#018a2cea-af05-4ae6-a8a1-33f4166205d6) 😀

Reply:
=> To be moved to a follow-up PR because a few left-overs will stay for now.